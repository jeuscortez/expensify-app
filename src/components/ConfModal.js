import React from 'react';
import {connect} from 'react-redux'
import {startRemoveExpense} from '../actions/expenses';
import {toggleModal} from '../actions/modal';
import Modal from 'react-modal';

export class ConfModal extends React.Component {
    constructor(props){
        super(props)
        this.handleCloseModal = this.handleCloseModal.bind(this);
        this.onRemoveOnHandleCloseModal = this.onRemoveOnHandleCloseModal.bind(this);
    }
    handleCloseModal () {
        this.props.toggleModal(this.props.showModal)
    }
    onRemoveOnHandleCloseModal () {
        this.props.toggleModal(this.props.showModal)
        this.props.startRemoveExpense({id: this.props.id})
        this.props.historyPush("/")
    }
    render(){
        return (
            <Modal
            isOpen={this.props.showModal}
            appElement={document.getElementById('app')}
            contentLabel="Are you sure you want to delete the expense?"
            onRequestClose={this.handleCloseModal}
            closeTimeoutMS={200}
            className="modal"
            >
                <h3 className="modal__title">Are you sure you want to delete the expense?</h3>
                <div className="modal-container">
                    <button className="button button--alert removeBtn" onClick={this.onRemoveOnHandleCloseModal}>Yes</button>
                    <button className="button button--secondary cancelBtn" onClick={this.handleCloseModal}>No</button>
                </div>
            </Modal>
        )
    }
};

const mapStateToProps = (state,props) => {
    return {
        showModal: state.modal.showModal
    }
};

const mapDispatchToProps = (dispatch,props) => ({
    startRemoveExpense: id=> dispatch(startRemoveExpense(id)),
    toggleModal: bool => dispatch(toggleModal(bool))
});

export default connect(mapStateToProps, mapDispatchToProps)(ConfModal);