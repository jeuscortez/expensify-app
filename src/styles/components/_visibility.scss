.show-for-mobile {
    @media (min-width: $desktop-breakpoint) {
        display: none;
    }
}

.show-for-desktop {
    @media(max-width: $desktop-breakpoint - .01rem) {//-.01rem secures to not apply both classes
        display: none;
    }
}