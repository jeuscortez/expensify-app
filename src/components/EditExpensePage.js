import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import { toggleModal } from '../actions/modal';
import ConfModal from './ConfModal';

export class EditExpensePage extends React.Component {
  constructor(props){
    super(props);
    this.handleOpenModal = this.handleOpenModal.bind(this);
  };
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  handleOpenModal () {
    this.props.toggleModal(this.props.showModal)
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
              <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.handleOpenModal}>Remove Expense</button>
        </div>
        <ConfModal 
          id={this.props.expense.id}
          historyPush={this.props.history.push}
        />
      </div>
    );
  }
};

// const mapStateToProps = (state, props) => ({
//   expense: state.expenses.find((expense) => expense.id === props.match.params.id)
// });

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => expense.id === props.match.params.id),
    showModal: state.modal.showModal
  }
};

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  //startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
  toggleModal: bool => dispatch(toggleModal(bool))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
