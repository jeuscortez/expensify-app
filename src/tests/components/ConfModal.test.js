import React from 'react';
import { shallow } from 'enzyme';
import { ConfModal } from '../../components/ConfModal';
import expenses from '../fixtures/expenses';

let id, historyPush, wrapper, toggleModal, startRemoveExpense;

beforeEach(() => {
        id = expenses[1].id
        historyPush = jest.fn()
        toggleModal = jest.fn()
        startRemoveExpense = jest.fn()
        wrapper = shallow(<ConfModal
          id={id}
          historyPush={historyPush}
          toggleModal={toggleModal}
          startRemoveExpense={startRemoveExpense}
        />)
      }
);

test("should render ConfModal correctly", () => {
    expect(wrapper).toMatchSnapshot()
});

test("should handle onClick cancelBtn", () => {
    wrapper.find(".cancelBtn").prop("onClick")()
    expect(toggleModal).toHaveBeenCalled()
    expect(toggleModal).toHaveBeenLastCalledWith(undefined)
});

test("should handle onClick removeBtn", () => {
    wrapper.find(".removeBtn").prop("onClick")()
    expect(startRemoveExpense).toHaveBeenCalled()
    expect(historyPush).toHaveBeenLastCalledWith("/")
    expect(startRemoveExpense).toHaveBeenCalledWith({id})
});