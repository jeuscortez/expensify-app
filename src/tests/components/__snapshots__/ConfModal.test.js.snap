// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfModal correctly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Are you sure you want to delete the expense?"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3
    className="modal__title"
  >
    Are you sure you want to delete the expense?
  </h3>
  <div
    className="modal-container"
  >
    <button
      className="button button--alert removeBtn"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button button--secondary cancelBtn"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;
